DM-2961: Port psfex support from HSC

mjuric opened the pull request.

natelust committed:
updated configure for single and double precision

natelust committed:
Moved the configuration into the build function.

This allows a cleaner code flow as the build must be run twice. Once for
double precision libraries and once for single precision

natelust committed:
Created a file to expose single precision library

mjuric commented on ups/eupspkg.cfg.sh:
```diff
@@ -1,4 +1,18 @@
 MAKE_INSTALL_TARGETS="-j1 install"
+
+config(){
+	clean_old_install
```
This usually goes into `install()` (see https://github.com/lsst/boost/blob/master/ups/eupspkg.cfg.sh)

natelust replied:
Changed

mjuric commented on ups/eupspkg.cfg.sh:
```diff
@@ -1,4 +1,18 @@
+
+install()
+{
+	build --enable-single
```
Watch out -- you're call the build function above again (in effect, rebuilding the package).

mjuric commented:
A general question about the changes to `eupspkg.cfg.sh` -- why did you need to override `config`, `build`, and `install`? I'd expect that setting `CONFIGURE_OPTIONS` and maybe `MAKE_INSTALL_FLAGS` would get the job done?

mjuric commented:
Is it that multi-job install on fftw has a bug (so you want to pass '-j1' to make?)

PaulPrice commented:
We need to configure, build and install fftw twice --- once for double and once for float, to produce the fftw and fftwf libraries.

mjuric commented:
Oy vey! That at least demands a 10-line comment block on the top of the file to explain what's going on :).

PaulPrice commented:
Should we, perhaps, implement this as a separate external package?

mjuric commented:
More on topic: I'd then propose to do the following instead:
* in prep, call `default_prep` to expand+patch the tarball, and then copy the result into two directories (e.g., sp/ and dp/).
* then override `config`, `build`, and `install` to just `cd` into those directories, do the configurations (as appropriate) and the installs.

I think that would be cleaner and more self-documenting.

natelust committed:
Removed unneeded config file

Exporting the single precision libraries was moved to the main fifth.cfg file

natelust committed:
Addition of single precision library

The single precision library included for export

natelust committed:
Reorganized eupspkg config script

Single and double precision libraries now build in their own source directories, though install to the same target.
Additional clean-ups to make the code more compliant with eupspkg scripts

add pwd for lsstsw debug

natelust commented:
I have adopted your recommendation with with two directory strategy. The source for single and double precision code can be found under source/sp and source/dp respectively. Each still installs to the same path

natelust committed:
Refined eupspkg to not depend on paths in prep

natelust committed:
Leave hidden files when clearning directory

natelust committed:
explicityly return to parent directory

natelust committed:
Add recursive option to rm

Fix up: removed logical not

natelust committed:
copy dp _build.log to parent for lsstsw

Lsstsw nessisary copy in if statement

natelust committed:
issues with if statements

natelust committed:
Changes to ensure lsstsw compatibility

Lsstsw has a paticular state the folder should be in such that it will
build the package repeatably. Issues existed where it would build fine
first, but fail on subsiquent builds. This was either due to directories
already being present, and or git directorys needing force rm'd.
